pkg/vfs/utils.go

```diff
@@ -0,0 +1,43 @@
+package vfs
+
+import (
+	"io/fs"
+	"os"
+
+	"encr.dev/pkg/eerror"
+)
+
+func FromDir(workingDir string, predicate func(fileName string) bool) (*VFS, error) {
```

Comment:
Might be useful to document the predicate only gets called for files and not directories? Is there any reason why it doesn't accept `(path string, info fs.DirEntry)`?